toolkit/scripts/check_signatures.py

```diff
@@ -20,20 +21,52 @@ def getSignature(fileName) -> str:
             sha256sum.update(read_data)
     return sha256sum.hexdigest()
 
-def check_folder(path):
+def find_file_and_check(path, filename, expected_signature) -> Optional[bool]:
+    path_to_check = os.path.join(path, filename)
+    if Path(path_to_check).is_file():
+        actual_signature = getSignature(path_to_check)
+        if actual_signature == expected_signature:
+            return True
+        print(f"ERROR: detected a mismatched signature for {filename}, expected [{expected_signature}] does not equal actual [{actual_signature}]")
+        return False
+
+    for content in os.listdir(path):
+        path_to_check = os.path.join(path, content)
+        if os.path.isdir(path_to_check):
+            result = find_file_and_check(path_to_check, filename, expected_signature)
+            if result is not None:
+                return result
+
+    return None
+
+def find_spec_folder_with_signatures_json(path: str) -> Optional[str]:
+    current = Path(path)
+    if Path(os.path.join(path, f"{current.name}.spec")).is_file():
+        if Path(os.path.join(path, f"{current.name}.signatures.json")).is_file():
+            return path
```

Comment:
this assumes that for SPECS/XXX/YYY.spec and SPECS/XXX/ZZZ.signatures.json, XXX==YYY and XXX==ZZZ.  hopefully this is true, it seems to be?

Reply:
Not quite. There can be multiple `.spec` files inside a folder (see: `golang` for instance). The expected structure is that for each `XXX.spec` there's an optional corresponding `XXX.signatures.json`. The signatures are optional, because some specs just don't have `Source` tags, so there are no sources to sign.

Reply:
ok ... will code something up with that in mind :)